samples/main.cpp

```diff
@@ -42,22 +42,29 @@ std::unique_ptr<NovelRT::Input::BasicInteractionRect> memeInteractionRect;
 
 int main(int argc, char *argv[])
 {
+  std::filesystem::path executableDirPath = NovelRT::Utilities::Misc::getExecutableDirPath();
+  std::filesystem::path resourcesDirPath = executableDirPath / "Resources";
+
+  std::filesystem::path fontsDirPath = resourcesDirPath / "Fonts";
+  std::filesystem::path imagesDirPath = resourcesDirPath / "Images";
+  std::filesystem::path scriptsDirPath = resourcesDirPath / "Scripts";
+
   //setenv("DISPLAY", "localhost:0", true);
   L = luaL_newstate();
   luaL_openlibs(L);
   lua_register(L, "average", average);
-  luaL_dofile(L, "avg.lua");
+  luaL_dofile(L, (scriptsDirPath / "avg.lua").string().c_str());
   lua_close(L);
   auto runner = NovelRT::NovelRunner(0, "NovelRTTest");
   auto console = NovelRT::LoggingService(NovelRT::Utilities::Misc::CONSOLE_LOG_APP);
 
   auto novelChanTransform = NovelRT::Transform(NovelRT::Maths::GeoVector<float>(1920 / 2, 1080 / 2), 2, NovelRT::Maths::GeoVector<float>(456, 618));
 
-  novelChanRect = runner.getRenderer()->createImageRect(novelChanTransform, 3, "novel-chan.png", NovelRT::Graphics::RGBAConfig(255, 0, 255, 255));
+  novelChanRect = runner.getRenderer()->createImageRect(novelChanTransform, 3, (imagesDirPath / "novel-chan.png").string(), NovelRT::Graphics::RGBAConfig(255, 0, 255, 255));
```

Comment:
it would probably be helpful if we took `std::filesystem::path` where we want a file path. It would avoid the need for consumers of the API to call `.string()`